src/main/integrations/main-process-session.ts

```diff
@@ -0,0 +1,51 @@
+import { getCurrentHub } from '@sentry/core';
+import { flush } from '@sentry/node';
+import { Integration, SessionStatus } from '@sentry/types';
+import { app } from 'electron';
+
+/** Adds Electron context to events and normalises paths. */
+export class MainProcessSession implements Integration {
+  /** @inheritDoc */
+  public static id: string = 'MainProcessSession';
+
+  /** @inheritDoc */
+  public name: string = MainProcessSession.id;
+
+  /** @inheritDoc */
+  public setupOnce(): void {
+    const hub = getCurrentHub();
+    hub.startSession();
+
+    this._addExitHandlerLast();
+
+    // We hook 'before-quit' and ensure our exit handler is still the last listener
+    app.on('before-quit', () => {
+      this._addExitHandlerLast();
+    });
```

Comment:
I'll add more detailed comments to explain this. 

On exit, Electron calls `before-quit` followed by `will-quit`. Developers can hook either of these and abort the exit via `event.preventDefault()`.

https://www.electronjs.org/docs/api/app#event-will-quit

The only way we can ensure that we don't interfere with developers custom exit logic, is to make sure our `will-quit` listener is last. Because `before-quit` is always called before `will-quit`, this seemed like a good enough time to ensure our listener is last.